src/Content/Widget.php

```diff
@@ -374,81 +374,59 @@ public static function categories($baseurl, $selected = '')
 	}
 
 	/**
-	 * Return common friends visitor widget
+	 * Show a random selection of five common contacts between the visitor and the viewed profile user.
 	 *
-	 * @param string $profile_uid uid
+	 * @param int    $uid      Viewed profile user ID
+	 * @param string $nickname Viewed profile user nickname
 	 * @return string|void
 	 * @throws \Friendica\Network\HTTPException\InternalServerErrorException
+	 * @throws \ImagickException
 	 */
-	public static function commonFriendsVisitor($profile_uid)
+	public static function commonFriendsVisitor(int $uid, string $nickname)
 	{
-		if (local_user() == $profile_uid) {
-			return;
-		}
-
-		$zcid = 0;
-
-		$cid = Session::getRemoteContactID($profile_uid);
-
-		if (!$cid) {
-			if (Profile::getMyURL()) {
-				$contact = DBA::selectFirst('contact', ['id'],
-						['nurl' => Strings::normaliseLink(Profile::getMyURL()), 'uid' => $profile_uid]);
-				if (DBA::isResult($contact)) {
-					$cid = $contact['id'];
-				} else {
-					$gcontact = DBA::selectFirst('gcontact', ['id'], ['nurl' => Strings::normaliseLink(Profile::getMyURL())]);
-					if (DBA::isResult($gcontact)) {
-						$zcid = $gcontact['id'];
-					}
-				}
-			}
+		if (local_user() == $uid) {
+			return '';
 		}
 
-		if ($cid == 0 && $zcid == 0) {
-			return;
+		$visitorPCid = local_user() ? Contact::getPublicIdByUserId(local_user()) : remote_user();
+		if (!$visitorPCid) {
+			return '';
 		}
 
-		if ($cid) {
-			$t = GContact::countCommonFriends($profile_uid, $cid);
-		} else {
-			$t = GContact::countCommonFriendsZcid($profile_uid, $zcid);
-		}
+		$localPCid = Contact::getPublicIdByUserId($uid);
 
-		if (!$t) {
-			return;
-		}
+		$condition = [
+			'NOT `self` AND NOT `blocked` AND NOT `hidden` AND `id` != ?',
+			$localPCid,
+		];
 
-		if ($cid) {
-			$r = GContact::commonFriends($profile_uid, $cid, 0, 5, true);
-		} else {
-			$r = GContact::commonFriendsZcid($profile_uid, $zcid, 0, 5, true);
+		$total = Contact\Relation::countCommon($localPCid, $visitorPCid, $condition);
+		if (!$total) {
+			return '';
 		}
 
-		if (!DBA::isResult($r)) {
-			return;
+		$commonContacts = Contact\Relation::listCommon($localPCid, $visitorPCid, $condition, 0, 5, true);
+		if (!DBA::isResult($commonContacts)) {
+			return '';
 		}
 
 		$entries = [];
-		foreach ($r as $rr) {
-			$contact = Contact::getByURL($rr['url']);
-			$entry = [
-				'url'   => Contact::magicLink($rr['url']),
-				'name'  => $contact['name'] ?? $rr['name'],
-				'photo' => Contact::getThumb($contact, $rr['photo']),
+		foreach ($commonContacts as $contact) {
+			$entries[] = [
+				'url'   => Contact::magicLink($contact['url']),
+				'name'  => $contact['name'],
+				'photo' => Contact::getThumb($contact, $contact['photo']),
```

Comment:
The second parameter to this function is only useful when it originates from another source than the array that is provided as first parameter.

Reply:
Got it, it used to fetch a contact from a contact and used the second parameter from the original contact, before it fetched the contact again.

Reply:
In fact at some point I want to review the usage of this function and possibly want to remove the second parameter. It is just needed in this transition phase.